app/helpers/schema_helpers.py

```diff
@@ -27,8 +28,17 @@ def wrapped_function(*args: Any, **kwargs: Any) -> Any:
         if not session_store:
             raise Unauthorized
 
-        session_data = session_store.session_data
-        schema = load_schema_from_session_data(session_data)
+        metadata = get_metadata(current_user)
+
+        language_code = (
+            session_store.session_data.language_code
+            if session_store.session_data
+            else None
+        )
```

Comment:
I would chain the if session data check with the if not session store check. If you didn't have session data, you wouldn't have metadata anyway. It would this make this clearer.

```py
        if not session_store or not session_store.session_data:
            raise Unauthorized

        metadata = get_metadata(current_user)
        language_code = session_store.session_data.language_code
```

We are missing a lot of issues because we haven't type checked the load schema methods. It's worth updating the methods the touches. i.e. here metadata won't be None but can be according to type hints and `load_schema_from_metadata` would fail if metadata was None.

Reply:
Updated this and added type hint